cns/service/main.go

```diff
@@ -187,6 +223,44 @@ func printVersion() {
 	fmt.Printf("Version %v\n", version)
 }
 
+// Try to register node with DNC when CNS is started in managed DNC mode
+func registerNode(httpRestService restserver.HTTPService, dncEP, infraVnet, nodeID string) {
+	logger.Printf("[Azure CNS] Registering node %s with Infrastructure Network: %s PrivateEndpoint: %s", nodeID, infraVnet, dncEP)
+
+	var (
+		numCPU   = runtime.NumCPU()
+		url      = fmt.Sprintf(acn.RegisterNodeURLFmt, dncEP, infraVnet, nodeID, numCPU, dncApiVersion)
+		response *http.Response
+		err      = fmt.Errorf("")
+		body     bytes.Buffer
+		httpc    = acn.GetHttpClient()
+	)
+
+	for sleep := true; err != nil; sleep = true {
+		response, err = httpc.Post(url, "application/json", &body)
+		if err == nil {
+			if response.StatusCode == http.StatusCreated {
+				var req cns.SetOrchestratorTypeRequest
+				json.NewDecoder(response.Body).Decode(&req)
+				httpRestService.SetNodeOrchestrator(&req)
+				sleep = false
+			} else {
+				logger.Errorf("[Azure CNS] Failed to register node with http status code %s", strconv.Itoa(response.StatusCode))
+			}
+
+			response.Body.Close()
+		} else {
+			logger.Errorf("[Azure CNS] Failed to register node with err: %+v", err)
```

Comment:
I had mentioned this in earlier review comments - 
Can you add infravnet guid and node id in the error messages in register and sync funcs? It will help when we are debugging the issue from kusto logs

Reply:
it's already there

logger.Printf("[Azure CNS] Registering node %s with Infrastructure Network: %s PrivateEndpoint: %s", nodeID, infraVnet, dncEP)

Reply:
I am talking about the error logs. Due to absence of that info in error log / log which says "Node Registered", we wont know what happened after 'Registering node'.